pkg/operation/hybridbotanist/cloud_config.go

```diff
@@ -55,12 +56,29 @@ func (b *HybridBotanist) generateCloudConfigChart() (*chartrenderer.RenderedChar
 		cloudProvider["config"] = cloudProviderConfig
 	}
 
+	machineTypes := b.Shoot.GetMachineTypesFromCloudProfile()
+	memoryThreshold, _ := resource.ParseQuantity("8Gi")
+
 	workers := []map[string]interface{}{}
-	for _, workerName := range userDataConfig.WorkerNames {
-		workers = append(workers, map[string]interface{}{
-			"name":       workerName,
-			"secretName": b.Shoot.ComputeCloudConfigSecretName(workerName),
-		})
+
+	for _, worker := range b.Shoot.GetWorkers() {
+		newWorker := map[string]interface{}{
+			"name":       worker.Name,
+			"secretName": b.Shoot.ComputeCloudConfigSecretName(worker.Name),
+		}
+		for _, machtype := range machineTypes {
+			if machtype.Name == worker.MachineType {
+				if machtype.Memory.Cmp(memoryThreshold) > 0 {
+					newWorker["evictionHard"] = "1Gi"
+					newWorker["evictionSoft"] = "1.5Gi"
+				} else {
+					newWorker["evictionHard"] = "5%"
+					newWorker["evictionSoft"] = "10%"
+				}
+				workers = append(workers, newWorker)
```

Comment:
Shouldn't this be moved to line 82? I see that the code would work as is but this `append` is quite confusing here. Also, the `break` is would no longer be needed after moving the statement (if there are multiple machine types with the same name then simply the last one would "win").

Reply:
Is it possible to have multiple machine types (different combination of cpu,memory, gpu, etc) with same name?
Even if it is, then the whole logic of comparing memory size values from a given type wouldn't work, right? And we could end up in situations where we reserve 6GB for 60GB machines (instead of 1.5GB). We would need to change the whole logic.
Also, `break` statement stops unnecessary comparisons once a match is found.
WDYT?

Reply:
> Is it possible to have multiple machine types (different combination of cpu,memory, gpu, etc) with same name?

Yes, there is no validation (yet?) that would prevent this. Agreed that allowing this does not make sense at all (also the quota validation logic would not work as expected). I will add this kind of validation separately.

Regarding the `append`: What I dislike is that it is only executed if the machine type is found, otherwise the worker configuration won't be added to the cloud-config. Let's assume the `machineTypes := b.Shoot.GetMachineTypesFromCloudProfile()` statement is buggy and returns an empty list - then, in the next reconciliation no `append` statements would be executed any more. We would end up in a situation without ANY worker configuration. If we would move the `append` below the `for`-loop we would "just" misconfigure the eviction parameters. I'm fine with the `break` to save unneeded iterations, but logically IMHO the `append` needs to be moved down.

Also, you have not yet added the new parameters to the `values.yaml` file of the Helm chart:
> Could you please add the new possible values (`eviction{Hard,Soft}MemoryAvailable`) to this section: https://github.com/gardener/gardener/blob/master/charts/shoot-cloud-config/charts/original/values.yaml#L22-L26 (to make clear what is configurable)?

I guess the PR is ready to be merged after these changes, so please squash the commits appropriately.

Thank you! 👍

Reply:
I have changed the logic such that there is always some default values provided for the eviction parameters. These values get replaced in the next `for` loop. This way, even if the statement `machineTypes := b.Shoot.GetMachineTypesFromCloudProfile()` is buggy, we will assign default values.
Does that make sense?